components/plugin_marketplace/marketplace_item/marketplace_item_plugin/marketplace_item_plugin.tsx

```diff
@@ -301,39 +365,54 @@ export default class MarketplaceItemPlugin extends React.PureComponent <Marketpl
                     </button>
                 </Link>
             );
-        }
-
-        let actionButton: JSX.Element;
-        if (this.props.error) {
-            actionButton = (
-                <FormattedMessage
-                    id='marketplace_modal.list.try_again'
-                    defaultMessage='Try Again'
-                />
-            );
         } else {
+            let actionLabel: JSX.Element;
+            if (this.props.error) {
+                actionLabel = (
+                    <FormattedMessage
+                        id='marketplace_modal.list.try_again'
+                        defaultMessage='Try Again'
+                    />
+                );
+            } else {
+                actionLabel = (
+                    <FormattedMessage
+                        id='marketplace_modal.list.install'
+                        defaultMessage='Install'
+                    />
+                );
+            }
+
+            let disableInstallButton = this.props.installing;
+            const limit = this.props.limits?.integrations?.enabled;
+            const usage = this.props.integrationsUsage?.enabled;
+            if (limit && usage && usage >= limit) {
```

Comment:
It is consistent with the current designs to disable the install button if the enabled integrations limit is hit, but I'm wondering if we want to do that.

@annelieseklein , should we allow installs of apps and plugins if the integrations limit is hit? An app or plugin can be installed (downloaded, but not active) without being enabled. We don't really enforce a download limit in any case (the user can just disable a plugin, install every plugin in the marketplace, and re-enable up to the limit).

To my understanding, we're ok with them having as many installed integrations as they want. This makes it easier for admins to switch between the plugins they want to use. It also would let admins install more core plugins whereas if we block download when max integrations are hit, the admin has to go through more steps if they want to install things like boards or playbooks (maybe they had previously deleted them and they aren't prebundled).

Reply:
> An app or plugin can be installed (downloaded, but not active) without being enabled. We don't really enforce a download limit in any case (the user can just disable a plugin, install every plugin in the marketplace, and re-enable up to the limit).

@neallred Even though the limit is actually centered around enabling integrations, I see value in disabling the "Install" button in this case. It's a strong reminder that they are in a restricted state. Since you wouldn't be able to enable that integration anyway, I don't see much value in installing it.

That being said, I'm 0/5 on my stance here.

> It also would let admins install more core plugins whereas if we block download when max integrations are hit, the admin has to go through more steps if they want to install things like boards or playbooks (maybe they had previously deleted them and they aren't prebundled).

I meant to include the check for the core plugins for this case, but I suppose I didn't because they aren't meant to be uninstalled. They will always be prebundled. I would rather allow those specific core plugins to be reinstalled, than use it as a reason to allow other plugins to be installed.

Reply:
I also think this makes sense since we moved to putting the emphasis on enabling. The toggle could just be disabled then. In that case I'd want to make sure there is a clear communication as to why it's disabled though. Makes me wonder if the banner shouldn't be dismissable then at the top @neallred would love to hear your thoughts on that.

Reply:
@neallred updated the design to reflect enabled install buttons and disabled toggle buttons upon limit reach - lmk your thoughts!

Reply:
Sounds good to me on leaving install buttons enabled and disabling further plugins from being disabled.

No strong opinion on whether the banner should be dismissable in light of this. I'd be ok leaving it as dismissable. If that proves problematic, we can start ignoring that preference and always show it.